RIOT-OS/RIOT
# gnrc_sixlowpan_frag_sfr: provide CongURE support

### Contribution description
This provides CongURE support for GNRC's 6LoWPAN SFR implementation and thus a to hook any CongURE congestion control implementation into SFR. The latter will be done in several follow-up PRs, this PR only adds the calling of the functions and tests to if the CongURE state machine is sane.



### Testing procedure
Tests are provided in `tests/gnrc_sixlowpan_frag_sfr_congure`. They should pass

```sh
BOARD="<your choice>" make -C tests/gnrc_sixlowpan_frag_sfr_congure -j flash test
```

They are based mostly on `tests/gnrc_sixlowpan_frag_sfr` with some cleanup (and tests that do not check sending behavior removed) and additional checks for CongURE. I tried to provide reasoning for the checks in the comments. Please read them carefully.



### Issues/PRs references
Depends on #16119 and #16133 and their respective dependencies.

![PR dependency graph](https://page.mi.fu-berlin.de/mlenders/congure.svg)
